app/uk/gov/hmrc/eoricommoncomponent/frontend/services/UserGroupIdSubscriptionStatusCheckService.scala

```diff
@@ -39,23 +39,21 @@ class UserGroupIdSubscriptionStatusCheckService @Inject() (
     save4Later.fetchCacheIds(groupId)
       .flatMap {
         case Some(cacheIds) =>
-          val sameUser    = cacheIds.internalId == internalId
           val sameService = cacheIds.serviceCode.contains(service.code)
 
           subscriptionStatusService
             .getStatus(idType, cacheIds.safeId.id)(hc, service, request)
-            .flatMap { status =>
-              if (status != SubscriptionProcessing)
+            .flatMap {
+              case NewSubscription | SubscriptionRejected | SubscriptionExists =>
                 if (sameService)
                   save4Later.deleteCachedGroupId(groupId).flatMap(_ => continue)
                 else
                   save4Later.deleteCacheIds(groupId).flatMap(_ => continue)
-              else
-                (sameUser, sameService) match {
-                  case (true, true)  => continue
-                  case (true, false) => userIsInProcess
-                  case (false, _)    => otherUserWithinGroupIsInProcess
-                }
+              case SubscriptionProcessing => //Processing is defined as 01, 11 or 14 subscriptionStatus code
+                if (cacheIds.internalId == internalId)
```

Comment:
Should we consider sameService here as per original implementation or is it safe not too? Can we remove the variable or is it used somewhere else?

Reply:
Also the sameUser could have multiple credId's in which case they could come through the  "otherUserIngroup" path. Liz had over 45 credIds I think. We have seen cases in production where users have been locked out of creating an enrolment on their preferred (current) credId/groupId pair because they already have an enrolment in a different guise (old/different credId/groupid pair). The same user with a credId in a different group can unassign themselves from an enrolment, but the enrolment will still belong to the (empty) group.... How users are uniquely identified inside EACD comes down to some undocumented magic which has never been explained. CredId/groupId is not it: https://jira.tools.tax.service.gov.uk/browse/ECC-1127